src/plan/markcompact/global.rs

```diff
@@ -0,0 +1,219 @@
+use super::gc_work::{
+    CalculateForwardingAddress, Compact, ForwardingProcessEdges, MarkingProcessEdges,
+    UpdateReferences,
+};
+use crate::mmtk::MMTK;
+use crate::plan::global::BasePlan;
+use crate::plan::global::CommonPlan;
+use crate::plan::global::GcStatus;
+use crate::plan::global::NoCopy;
+use crate::plan::markcompact::mutator::ALLOCATOR_MAPPING;
+use crate::plan::AllocationSemantics;
+use crate::plan::Plan;
+use crate::plan::PlanConstraints;
+use crate::policy::markcompactspace::MarkCompactSpace;
+use crate::policy::space::Space;
+use crate::scheduler::gc_work::*;
+use crate::scheduler::*;
+use crate::util::alloc::allocators::AllocatorSelector;
+#[cfg(not(feature = "global_alloc_bit"))]
+use crate::util::alloc_bit::ALLOC_SIDE_METADATA_SPEC;
+use crate::util::heap::layout::heap_layout::Mmapper;
+use crate::util::heap::layout::heap_layout::VMMap;
+use crate::util::heap::layout::vm_layout_constants::{HEAP_END, HEAP_START};
+use crate::util::heap::HeapMeta;
+use crate::util::heap::VMRequest;
+use crate::util::metadata::side_metadata::{SideMetadataContext, SideMetadataSanity};
+use crate::util::opaque_pointer::*;
+use crate::util::options::UnsafeOptionsWrapper;
+use crate::vm::VMBinding;
+use enum_map::EnumMap;
+use std::sync::Arc;
+
+pub struct MarkCompact<VM: VMBinding> {
+    pub mc_space: MarkCompactSpace<VM>,
+    pub common: CommonPlan<VM>,
+}
+
+pub const MARKCOMPACT_CONSTRAINTS: PlanConstraints = PlanConstraints {
+    moves_objects: true,
+    gc_header_bits: 2,
+    gc_header_words: 1,
+    num_specialized_scans: 2,
+    // needs_forward_after_liveness: true, // enable this once finalizable_processor can work with markcompact
```

Comment:
What is the status for this? From the code, it seems that you are using `ForwardFinalization` (which implies `needs_forward_after_liveness == true`).

Reply:
I did not use ```schedule_common``` so I do not need ```needs_forward_after_liveness```. Once ```schedule_common``` can work with markcompact, we should enable this.

Reply:
I would suggest still putting this as `true`, as it is true (Unless changing it to true will break something). Can you do this change, and also merge with master?

Reply:
okay